aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -11,6 +11,43 @@
 namespace at {
 namespace native {
 
+Tensor quantize_per_tensor_dynamic(
+    const Tensor& self,
+    ScalarType dtype,
+    bool reduce_range) {
+  TORCH_CHECK( (dtype == ScalarType::QInt8 || dtype == ScalarType::QUInt8), "dtype ", dtype, "not supported");
+  auto input_contig = self.contiguous();
+  float x_min = input_contig.min().item<float>();
+  float x_max = input_contig.max().item<float>();
+
+  if (reduce_range && at::globalContext().qEngine() == at::QEngine::QNNPACK) {
+    reduce_range = false;
+  }
+
+  int qmin = 0;
+  int qmax = 255;
+
+  // should we check additional QInt types beyond 8 bits?
+  if (dtype == ScalarType::QInt8) {
+    qmin = -128;
+    qmax = 127;
+  } else if (dtype == ScalarType::QUInt8) {
+    qmin = 0;
+    qmax = 255;
+  }
```

Comment:
nit: we can add a comment here saying that we should add new branches when we add support for other dtypes

Reply:
Done